hack/loadtest/util/help.go

```diff
@@ -167,15 +181,116 @@ func CleanupAll(hubClient client.Client) error {
-		klog.ErrorS(err, "failed to list namespace")
+	return nil
+}
+
+func executeStrategy(crp *v1beta1.ClusterResourcePlacement, obj *unstructured.Unstructured) error {
+	strategy, foundType, err := unstructured.NestedString(obj.Object, "spec", "strategy", "type")
+	if err != nil {
+		klog.ErrorS(err, "Failed to get strategy.")
 		return err
 	}
-	for index, crp := range crps.Items {
-		if err := hubClient.Delete(context.Background(), &crps.Items[index]); err != nil {
-			klog.ErrorS(err, "failed to delete crp", "crp", crp.Name)
-		}
+	if !foundType {
+		return nil
+	}
+	crp.Spec.Strategy.Type = v1beta1.RolloutStrategyType(strategy)
+
+	rollingUpdate, found, err := unstructured.NestedMap(obj.Object, "spec", "strategy", "rollingUpdate")
+	if err != nil {
+		klog.ErrorS(err, "Failed to get RollingUpdate.")
+		return err
+	}
+	if !found {
+		klog.Info("RollingUpdate not found in file.")
+		return nil
+	}
+	crp.Spec.Strategy.RollingUpdate = &v1beta1.RollingUpdateConfig{}
+
+	if maxUnavailable, found := rollingUpdate["maxUnavailable"].(string); found {
+		maxUnvailable := intstr.FromString(maxUnavailable)
+		crp.Spec.Strategy.RollingUpdate.MaxUnavailable = &maxUnvailable
+	}
+	if maxSurge, found := rollingUpdate["maxSurge"].(string); found {
+		maxSurge := intstr.FromString(maxSurge)
+		crp.Spec.Strategy.RollingUpdate.MaxSurge = &maxSurge
+	}
+	if unavailablePeriodSeconds, found := rollingUpdate["unavailablePeriodSeconds"].(int64); found {
+		crp.Spec.Strategy.RollingUpdate.UnavailablePeriodSeconds = pointer.Int(int(unavailablePeriodSeconds))
 	}
 	return nil
 }
+
+func executeRevisionHistoryLimit(crp *v1beta1.ClusterResourcePlacement, obj *unstructured.Unstructured) error {
+	revisionLimit, found, err := unstructured.NestedInt64(obj.Object, "spec", "revisionHistoryLimit")
+	if err != nil {
+		klog.ErrorS(err, "Failed to get RevisionHistoryLimit.")
+		return err
+	}
+	if !found {
+		return nil
+	}
+	crp.Spec.RevisionHistoryLimit = pointer.Int32(int32(revisionLimit))
+	return nil
+}
+
+func executePickFixedPolicy(crp *v1beta1.ClusterResourcePlacement, obj *unstructured.Unstructured, clusterNames ClusterNames) (ClusterNames, error) {
+	names, found, err := unstructured.NestedStringSlice(obj.Object, "spec", "policy", "clusterNames")
+	if err != nil {
+		klog.ErrorS(err, "Failed to get cluster names.")
+		return clusterNames, err
+	}
+	if !found {
+		klog.Infof("Cluster Names not found in file.")
+		return clusterNames, nil
+	}
+
+	for _, name := range names {
+		if err := clusterNames.Set(name); err != nil {
+			klog.ErrorS(err, "Error setting cluster name: %v")
+			return clusterNames, err
+		}
+	}
+
+	crp.Spec.Policy = &v1beta1.PlacementPolicy{
+		PlacementType: v1beta1.PickFixedPlacementType,
+		ClusterNames:  names,
+	}
+	return clusterNames, nil
+}
+
+func applyCRP(crp *v1beta1.ClusterResourcePlacement, crpFile string, nsName string, clusterNames ClusterNames) error {
+	obj, err := readObjFromFile(fmt.Sprintf("hack/loadtest/%s", crpFile), nsName)
```

Comment:
Hi Britania! I couldn't help noticing that many functions here deal with unstructured objects; a type cast is possible, is there any particular reason why unstructured must be used?

Reply:
Typically [DefaultUnstructuredConverter](https://pkg.go.dev/k8s.io/apimachinery/pkg/runtime#pkg-variables) can be used for this purpose.